cmd/tailscale/cli/update.go

```diff
@@ -209,46 +221,24 @@ func (up *updater) updateSynology() error {
 func (up *updater) updateDebLike() error {
 	ver := updateArgs.version
 	if ver == "" {
-		res, err := http.Get("https://pkgs.tailscale.com/" + up.track + "/?mode=json")
+		var err error
+		ver, err = latestTailscaleVersion(up.track)
 		if err != nil {
 			return err
 		}
-		var latest struct {
-			Tarballs map[string]string // ~goarch (ignoring "geode") => "tailscale_1.34.2_mips.tgz"
-		}
-		err = json.NewDecoder(res.Body).Decode(&latest)
-		res.Body.Close()
-		if err != nil {
-			return fmt.Errorf("decoding JSON: %v: %w", res.Status, err)
-		}
-		f, ok := latest.Tarballs[runtime.GOARCH]
-		if !ok {
-			return fmt.Errorf("can't update architecture %q", runtime.GOARCH)
-		}
-		ver, _, ok = strings.Cut(strings.TrimPrefix(f, "tailscale_"), "_")
-		if !ok {
-			return fmt.Errorf("can't parse version from %q", f)
-		}
 	}
 	if up.currentOrDryRun(ver) {
 		return nil
 	}
 
-	track := "unstable"
-	if stable, ok := versionIsStable(ver); !ok {
-		return fmt.Errorf("malformed version %q", ver)
-	} else if stable {
-		track = "stable"
-	}
```

Comment:
this is already handled here: https://github.com/tailscale/tailscale/blob/cb0e3b4497426d21a4d705a01ee0b6f140aa674f/cmd/tailscale/cli/update.go#L118-L132